File: AC/tests/benchmark_suite.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/**
 * Benchmark Suite — C implementation
 * Same algorithms as AC benchmark_suite.ac
 */

long fib(long n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

long sum_arr(long* arr, int len) {
    long total = 0;
    for (int i = 0; i < len; i++) {
        total += arr[i];
    }
    return total;
}

double mean(long* arr, int len) {
    if (len == 0) return 0;
    return (double)sum_arr(arr, len) / len;
}

long distance(long* arr, int len) {
    if (len == 0) return 0;
    long minv = arr[0];
    long maxv = arr[0];
    for (int i = 0; i < len; i++) {
        if (arr[i] < minv) minv = arr[i];
        if (arr[i] > maxv) maxv = arr[i];
    }
    return maxv - minv;
}

double stddev(long* arr, int len) {
    if (len == 0) return 0;
    double m = mean(arr, len);
    double sum_sq = 0;
    for (int i = 0; i < len; i++) {
        double diff = arr[i] - m;
        sum_sq += diff * diff;
    }
    return sum_sq / len;
}

int sieve(int n, int* result) {
    if (n < 2) return 0;

    int* prime = (int*)malloc((n + 1) * sizeof(int));
    for (int i = 0; i <= n; i++) {
        prime[i] = 1;
    }
    prime[0] = prime[1] = 0;

    for (int p = 2; p * p <= n; p++) {
        if (prime[p]) {
            for (int i = p * p; i <= n; i += p) {
                prime[i] = 0;
            }
        }
    }

    int count = 0;
    for (int i = 2; i <= n; i++) {
        if (prime[i]) {
            result[count++] = i;
        }
    }

    free(prime);
    return count;
}

long loop_work() {
    long total = 0;
    for (int i = 0; i < 100; i++) {
        total += i;
    }
    return total;
}

double array_ops() {
    long arr[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int len = 10;
    long s = sum_arr(arr, len);
    double m = mean(arr, len);
    double std = stddev(arr, len);
    long dist = distance(arr, len);
    return s + m + std + dist;
}

int main() {
    long fib_result = fib(28);
    long loop_result = loop_work();

    int* sieve_primes = (int*)malloc(51 * sizeof(int));
    int sieve_count = sieve(50, sieve_primes);
    free(sieve_primes);

    double array_result = array_ops();

    long checksum = fib_result + loop_result + sieve_count;
    checksum = (long)(checksum + array_result);

    printf("%ld\n", checksum);
    return 0;
}

File: AC/tests/benchmark_suite.go
package main

import (
	"fmt"
)

/**
 * Benchmark Suite — Go implementation
 * Same algorithms as AC benchmark_suite.ac
 */

func fib(n int64) int64 {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func sumArr(arr []int64) int64 {
	total := int64(0)
	for _, i := range arr {
		total += i
	}
	return total
}

func mean(arr []int64) float64 {
	if len(arr) == 0 {
		return 0
	}
	return float64(sumArr(arr)) / float64(len(arr))
}

func distance(arr []int64) int64 {
	if len(arr) == 0 {
		return 0
	}
	minv := arr[0]
	maxv := arr[0]
	for _, i := range arr {
		if i < minv {
			minv = i
		}
		if i > maxv {
			maxv = i
		}
	}
	return maxv - minv
}

func stddev(arr []int64) float64 {
	if len(arr) == 0 {
		return 0
	}
	m := mean(arr)
	sumSq := 0.0
	for _, i := range arr {
		diff := float64(i) - m
		sumSq += diff * diff
	}
	return sumSq / float64(len(arr))
}

func sieve(n int) []int {
	if n < 2 {
		return []int{}
	}

	prime := make([]bool, n+1)
	for i := 0; i <= n; i++ {
		prime[i] = true
	}
	prime[0] = false
	prime[1] = false

	for p := 2; p*p <= n; p++ {
		if prime[p] {
			for i := p * p; i <= n; i += p {
				prime[i] = false
			}
		}
	}

	result := []int{}
	for i := 2; i <= n; i++ {
		if prime[i] {
			result = append(result, i)
		}
	}

	return result
}

func loopWork() int64 {
	total := int64(0)
	for i := 0; i < 100; i++ {
		total += int64(i)
	}
	return total
}

func arrayOps() float64 {
	arr := []int64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
	s := sumArr(arr)
	m := mean(arr)
	std := stddev(arr)
	dist := distance(arr)
	return float64(s) + m + std + float64(dist)
}

func main() {
	fibResult := fib(28)
	loopResult := loopWork()
	sieveResult := len(sieve(50))
	arrayResult := arrayOps()

	checksum := fibResult + int64(loopResult) + int64(sieveResult)
	checksum = int64(float64(checksum) + arrayResult)

	fmt.Println(checksum)
}
